app/src/main/java/com/demoapp/viewmodel/MainActivityViewModel.kt

```diff
@@ -25,22 +25,26 @@ class MainActivityViewModel @Inject constructor(private val dataRepository: Data
     }
 
     init {
-        populateData()
+        checkIfNeedToShoLoader()
+        getDataFromApi()
+    }
 
-        if (_allCategoryList.value.isNullOrEmpty())
-            _showLoader.value = true
 
+    private fun checkIfNeedToShoLoader() {
         viewModelScope.launch {
-            dataRepository.getCategoryData()
-            _showLoader.value = false
-            populateData()
+            if (categoryList.first().isNullOrEmpty())
+                _showLoader.value = true
         }
     }
 
-    private fun populateData() {
+    private fun getDataFromApi() {
         viewModelScope.launch {
-            val resultList = dataRepository.getLocalData()
-            _allCategoryList.value = resultList
+            val res = dataRepository.getCategoryData()
+            if (res != null) {
+                dataRepository.insertData(res)
+            }
```

Comment:
do this triggers from Repository itself

Reply:
Created DataSource for transactions with local database